### PR TITLE
Added quoting around compile properties

### PR DESCRIPTION
**Pull request by judovana**
Hello!

We are hitting:
```
**************************************************************************

System.getProperty('java.vm.name')=OpenJDK 64-Bit Server VM

System.getProperty('java.vendor')=Undefined Vendor

Warning: cannot determine vendor, use System.getProperty('java.vendor')=Undefined Vendor directly.

System.getProperty('os.name')=Linux

System.getProperty('os.arch')=aarch64

System.getProperty('java.fullversion')=null

System.getProperty('sun.arch.data.model')=64

make -f clean.mk cleanBuild
make[1]: Entering directory '/home/tester/aqa-crypto-tests/aqa-tests/TKG'
rm -f -r /home/tester/aqa-crypto-tests/aqa-tests/TKG/../../jvmtest
make[1]: Leaving directory '/home/tester/aqa-crypto-tests/aqa-tests/TKG'
make -f compile.mk compile
make[1]: Entering directory '/home/tester/aqa-crypto-tests/aqa-tests/TKG'
rm -f -r /home/tester/aqa-crypto-tests/aqa-tests/TKG/../TKG/output_compilation; \
mkdir -p /home/tester/aqa-crypto-tests/aqa-tests/TKG/../TKG/output_compilation; \
(ant -f scripts/build_test.xml -DTEST_ROOT=/home/tester/aqa-crypto-tests/aqa-tests/TKG/.. -DBUILD_ROOT=/home/tester/aqa-crypto-tests/aqa-tests/TKG/../../jvmtest -DJDK_VERSION=21 -DJDK_IMPL=hotspot -DJDK_VENDOR=Undefined Vendor -DJCL_VERSION=latest -DBUILD_LIST=functional/security/Crypto -DRESOURCES_DIR=/home/tester/aqa-crypto-tests/aqa-tests/TKG/../../jvmtest/TestConfig/resources -DSPEC=linux_aarch64 -DTEST_JDK_HOME=/usr/lib/jvm/java-21-openjdk-jdk.21.0.4.5-0.ojdk21~u~upstream.hotspot.fastdebug.sdk.el8.aarch64.tarxz -DJVM_VERSION=openjdk21 -DLIB_DIR=/tmp/tmp.CgKlo9A1Qu  2>&1; echo $? ) | tee "/home/tester/aqa-crypto-tests/aqa-tests/TKG/../TKG/output_compilation/compilation.log"; \
if [ -z "$(tail -1 /home/tester/aqa-crypto-tests/aqa-tests/TKG/../TKG/output_compilation/compilation.log | grep 0)" ]; then perl scripts/moveDmp.pl --compileLogPath="/home/tester/aqa-crypto-tests/aqa-tests/TKG/../TKG/output_compilation/compilation.log" --testRoot="/home/tester/aqa-crypto-tests/aqa-tests/TKG/.."; false; else rm -f -r "/home/tester/aqa-crypto-tests/aqa-tests/TKG/../TKG/output_compilation"; fi
Picked up JAVA_TOOL_OPTIONS: -Djava.security.manager=allow
Buildfile: /home/tester/aqa-crypto-tests/aqa-tests/TKG/scripts/build_test.xml
     [echo]  build.list is TestConfig/build.xml,functional/security/Crypto/build.xml 

BUILD FAILED
Target "Vendor" does not exist in the project "Compile Java Test Projects". 

Total time: 0 seconds
1
make[1]: Leaving directory '/home/tester/aqa-crypto-tests/aqa-tests/TKG'
make[1]: *** [compile.mk:45: compile] Error 1
make: *** [makefile:87: compile] Error 2
```

Which suggests that under some circumstances vendor with spaces is not handled properly. THis PR may be very wrong,  the call from aqa-tests may be guilty., BUt I have hard times to manually trigger it, however:

In case of rpms vendor is: "Red Hat, Inc."
Seems like some logic simplifies it to "redhat" and it is then passed as that:
ant -f scripts/build_test.xml ... -DJDK_VENDOR=redhat ...

In case undef builds seems to be: "Undefined Vendor"
which it says is unrecognized and passed literally:
ant -f scripts/build_test.xml ... -DJDK_VENDOR=Undefined Vendor ...

### CODE REVIEW

**Comment by judovana** [comment]
We were able to workaround it by 
```
  export JDK_VENDOR="redhat"
  export AUTO_DETECT="false"

```

But I stil think it should be quoted, thoughts?

**Comment by smlambert** [comment]
Happy to bring this change in (with some additional testing expected), fixes a "closed as won't fix" issue https://github.com/adoptium/TKG/issues/502

File: compile.mk [hunk 1]
```diff
@@ -38,7 +38,7 @@ TEST_FLAG_PARAM := -DTEST_FLAG=$(TEST_FLAG)
 else
 TEST_FLAG_PARAM :=
 endif
-COMPILE_CMD=ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJDK_VENDOR=$(JDK_VENDOR) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${COMPILE_BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DSPEC=${SPEC} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION) -DLIB_DIR=$(LIB_DIR) ${TEST_FLAG_PARAM}
+COMPILE_CMD=ant -f scripts$(D)build_test.xml -DTEST_ROOT="$(TEST_ROOT)" -DBUILD_ROOT="$(BUILD_ROOT)" -DJDK_VERSION="$(JDK_VERSION)" -DJDK_IMPL="$(JDK_IMPL)" -DJDK_VENDOR="$(JDK_VENDOR)" -DJCL_VERSION="$(JCL_VERSION)" -DBUILD_LIST="${COMPILE_BUILD_LIST}" -DRESOURCES_DIR="${RESOURCES_DIR}" -DSPEC="${SPEC}" -DTEST_JDK_HOME="${TEST_JDK_HOME}" -DJVM_VERSION="$(JVM_VERSION)" -DLIB_DIR="$(LIB_DIR)" ${TEST_FLAG_PARAM}
```
Comment (sophia-guo) [inline compile.mk:41]: Could we only quote JDK_VENDOR or non directory one unless this can be tested in windows platforms (jenkins or local)?

Comment (LongyuZhang) [reply compile.mk:41]: Also, we may need to use `$(Q)` to replace `"` and test it works on all platforms.

**Comment by judovana** [comment]
Sure. TY! Will elebaorate

**Comment by judovana** [comment]
Jsut a question - I woudl thought, that the directories one would be the most important, due to possible space in path... But seeing the https://github.com/adoptium/TKG/issues/502 I smell troubles :)

**Comment by sophia-guo** [comment]
yes, feel like directory ones would be the most possible ones with spaces in path, especially for windows and mac. However we haven't got problems by now. The reason we might need to be cautious is quotes might be added later somewhere else. Then there will be extra quotes. Though if tests on windows and mac are fine then we are good to do that.

**Comment by judovana** [comment]
```
$ git checkout vendorWithSpaces && JDK_VENDOR="part1 part2" AUTO_DETECT=false TEST_JDK_HOME=/usr/lib/jvm/java-11-openjdk make 
Already on 'vendorWithSpaces'
Your branch is up to date with 'origin/vendorWithSpaces'.
JAVA_HOME is set to /usr/lib/jvm/java-11-openjdk
UNAME_OS is Linux
LIB_DIR is set to /home/jvanek/git/TKG/../TKG/lib
rm -f -r /home/jvanek/git/TKG/../TKG/output_compilation; \
mkdir -p /home/jvanek/git/TKG/../TKG/output_compilation; \
(ant -f ./scripts/build_tools.xml -DTEST_JDK_HOME=/usr/lib/jvm/java-11-openjdk -DTEST_ROOT=/home/jvanek/git/TKG/.. -DLIB_DIR=/home/jvanek/git/TKG/../TKG/lib 2>&1; echo $? ) | tee "/home/jvanek/git/TKG/../TKG/output_compilation/compilation.log"; \
if [ -z "$(tail -1 /home/jvanek/git/TKG/../TKG/output_compilation/compilation.log | grep 0)" ]; then perl scripts/moveDmp.pl --compileLogPath="/home/jvanek/git/TKG/../TKG/output_compilation/compilation.log" --testRoot="/home/jvanek/git/TKG/.."; false; else rm -f -r "/home/jvanek/git/TKG/../TKG/output_compilation"; fi
...
System.getProperty('java.vendor')=Red Hat, Inc.
...
mkdir -p /home/jvanek/git/TKG/../TKG/output_compilation; \
(ant -f scripts/build_test.xml -DTEST_ROOT=/home/jvanek/git/TKG/.. -DBUILD_ROOT=/home/jvanek/git/TKG/../../jvmtest -DJDK_VERSION=11 -DJDK_IMPL=hotspot "-DJDK_VENDOR=part1 part2" -DJCL_VERSION=latest -DBUILD_LIST= -DRESOURCES_DIR=/home/jvanek/git/TKG/../../jvmtest/TestConfig/resources -DSPEC=linux_x86-64 -DTEST_JDK_HOME=/usr/lib/jvm/java-11-openjdk -DJVM_VERSION=openjdk11 -DLIB_DIR=/home/jvanek/git/TKG/../TKG/lib  2>&1; echo $? ) | tee "/home/jvanek/git/TKG/../TKG/output_compilation/compilation.log"; \
if [ -z "$(tail -1 /home/jvanek/git/TKG/../TKG/output_compilation/compilation.log | grep 0)" ]; then perl scripts/moveDmp.pl --compileLogPath="/home/jvanek/git/TKG/../TKG/output_compilation/compilation.log" --testRoot="/home/jvanek/git/TKG/.."; false; else rm -f -r "/home/jvanek/git/TKG/../TKG/output_compilation"; fi
Buildfile: /home/jvanek/git/TKG/scripts/build_test.xml
     [echo]  build.list is */build.xml 

-create_test_directory:
    [mkdir] Created dir: /home/jvanek/jvmtest

stage_test_material:
     [copy] Copying 134 files to /home/jvanek/jvmtest

build:

-pre-init:

....
```

```
git checkout master  && JDK_VENDOR="part1 part2" AUTO_DETECT=false TEST_JDK_HOME=/usr/lib/jvm/java-11-openjdk make 
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
JAVA_HOME is set to /usr/lib/jvm/java-11-openjdk
UNAME_OS is Linux
LIB_DIR is set to /home/jvanek/git/TKG/../TKG/lib
...
System.getProperty('java.vendor')=Red Hat, Inc.
...
mkdir -p /home/jvanek/git/TKG/../TKG/output_compilation; \
(ant -f scripts/build_test.xml -DTEST_ROOT=/home/jvanek/git/TKG/.. -DBUILD_ROOT=/home/jvanek/git/TKG/../../jvmtest -DJDK_VERSION=11 -DJDK_IMPL=hotspot -DJDK_VENDOR=part1 part2 -DJCL_VERSION=latest -DBUILD_LIST= -DRESOURCES_DIR=/home/jvanek/git/TKG/../../jvmtest/TestConfig/resources -DSPEC=linux_x86-64 -DTEST_JDK_HOME=/usr/lib/jvm/java-11-openjdk -DJVM_VERSION=openjdk11 -DLIB_DIR=/home/jvanek/git/TKG/../TKG/lib  2>&1; echo $? ) | tee "/home/jvanek/git/TKG/../TKG/output_compilation/compilation.log"; \
if [ -z "$(tail -1 /home/jvanek/git/TKG/../TKG/output_compilation/compilation.log | grep 0)" ]; then perl scripts/moveDmp.pl --compileLogPath="/home/jvanek/git/TKG/../TKG/output_compilation/compilation.log" --testRoot="/home/jvanek/git/TKG/.."; false; else rm -f -r "/home/jvanek/git/TKG/../TKG/output_compilation"; fi
Buildfile: /home/jvanek/git/TKG/scripts/build_test.xml
     [echo]  build.list is */build.xml 

BUILD FAILED
Target "part2" does not exist in the project "Compile Java Test Projects". 

Total time: 0 seconds
1
make[1]: *** [compile.mk:45: compile] Error 1
make[1]: Leaving directory '/home/jvanek/git/TKG'
make: *** [makefile:87: compile] Error 2

```

**Review by smlambert — APPROVED** [review]

**Review by LongyuZhang — APPROVED** [review]
LGTM.

**Comment by judovana** [comment]
tyvm!

**Pull request merged**